### PR TITLE
use idenity encoding on client if no compression features are enabled

### LINKED ISSUE #1731: Disable compression headers when disabling compression feature
## Expected Behavior
I expect awc to set the Accept-Encoding header to `identity` when the compression feature in actix-web is disabled.

## Current Behavior
The Accept-Encoding header is set to `br`. Any HTTP response from a webserver that makes use of the header will send brotli-encoded bodies, which awc can not decode because the feature is disabled, resulting in a "useless" body for the user.

## Possible Solution
See expected behavior. I am unfamiliar with the source, but https://github.com/actix/actix-web/blob/master/awc/src/request.rs#L24 and below seems like the culprit.

## Steps to Reproduce (for bugs)
This code snippet does it just fine. The web server that caused me to run into the issue requires authentication, so I cannot post that here.
```rust
#[actix_web::main]
async fn main() {
    let client = actix_web::client::Client::new();
    let response = client
        .get("https://httpbin.org/anything")
        .send()
        .await
        .unwrap()
        .body()
        .await
        .unwrap();
    // {
    //  "args": {},
    //  "data": "",
    //  "files": {},
    //  "form": {},
    //  "headers": {
    //    "Accept-Encoding": "br",
    //    "Content-Length": "0",
    //    "Host": "httpbin.org",
    //    "X-Amzn-Trace-Id": "Root=1-5f833c44-343c4ab909998d701ad5df2d"
    //  },
    //  "json": null,
    //  "method": "GET",
    //  "origin": "not-gonna-leak-my-ip",
    //   "url": "https://httpbin.org/anything"
    // }
    println!("{}", String::from_utf8(response.to_vec()).unwrap());
}
```

I was able to fix it manually with 
```rust
                    .header(
                        header::ACCEPT_ENCODING,
                        HeaderValue::from_static("identity"),
                    )
```
but this should not be the default with these feature flags.

## Your Environment

* Rust Version (I.e, output of `rustc -V`): rustc 1.48.0-nightly (7402a3944 2020-09-13)
* Actix Web Version: 3.1.0, `default-features = false` to disable compression

### PR DESCRIPTION
**Pull request by j-brn**
## PR Type
Bugfix

## Overview
Fixes https://github.com/actix/actix-web/issues/1731 by using `Accept-Encoding: identity` when no compression feature is enabled.

### CODE REVIEW

File: awc/src/request.rs [hunk 1]
```diff
@@ -23,8 +23,17 @@ use crate::ClientConfig;
...
+    not(any(feature = "flate2-zlib", feature = "flate2-rust")),
+    feature = "compress"
+))]
 const HTTPS_ENCODING: &str = "br";
+#[cfg(not(any(
+    feature = "flate2-zlib",
+    feature = "flate2-rust",
+    feature = "compress"
+)))]
+const HTTPS_ENCODING: &str = "identity";
```
Comment (robjtede) [inline awc/src/request.rs]: would it be clearer to use `cfg-if` ?

Comment (j-brn) [reply awc/src/request.rs]: Do you mean https://docs.rs/cfg-if/1.0.0/cfg_if ? I can't find anything like `cfg-it` it under https://doc.rust-lang.org/reference/conditional-compilation.html

Comment (j-brn) [reply awc/src/request.rs]: That looks nice indeed and its already in the dependency tree. I'll change it.

Comment (robjtede) [reply awc/src/request.rs]: should avoid any situations where a combination of feature flags causes compile errors

Comment (j-brn) [reply awc/src/request.rs]: Done. The solution i originally had i mind doesn't work with the macro unfortunately, but it allows to reduce the complexity of the cfg attributes, so its still a win i guess.
```rust
const HTTPS_ENCODING: &str = cfg_if::cfg_if! {
    if #[cfg(any(feature = "flate2-zlib", feature = "flate2-rust"))] {
         "br, gzip, deflate";
    } else if #[cfg(feature = "compress")] {
        "br";
    } else {
        "identity";
    }
};
```

**Review by JohnTitor — APPROVED** [review]
Looks good, could you add an entry to the changelog as this changes the current behavior as a bug fix?

**Comment by j-brn** [comment]
> 
> 
> Looks good, could you add an entry to the changelog as this changes the current behavior as a bug fix?

Added in c2aeea512d4770370f80423d4217bc5074f8b09c

File: CHANGES.md [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
-* Implement Logger middleware regex exclude pattern [#1723]
 
+* Implement Logger middleware regex exclude pattern [#1723]
 [#1723]: https://github.com/actix/actix-web/pull/1723
 
+### Fixed
+
+* awc now uses `Accept-Encoding: identity` instead of `Accept-Encoding: br` when no compression feature is enabled
+[#1737](https://github.com/actix/actix-web/pull/1737)
+
```
Comment (robjtede) [inline CHANGES.md]: this should be in `awc/CHANGEs.md`

File: CHANGES.md [hunk 3]
```diff
@@ -1,10 +1,15 @@
 # Changes
 
 ## Unreleased - 2020-xx-xx
-* Implement Logger middleware regex exclude pattern [#1723]
 
+* Implement Logger middleware regex exclude pattern [#1723]
```
Comment (robjtede) [inline CHANGES.md]: undo the diff noise

File: CHANGES.md (hunk 2 above)
Comment (j-brn) [reply CHANGES.md]: -> 06841e111c58a30537455fb2e0ba8e72df0ffcfe

File: CHANGES.md (hunk 3 above)
Comment (j-brn) [reply CHANGES.md]: -> 06841e111c58a30537455fb2e0ba8e72df0ffcfe

**Review by robjtede — APPROVED** [review]

**Pull request merged**